test/e2e/util.go

```diff
@@ -401,6 +407,67 @@ func waitForPodsRunningReady(ns string, minPods int, timeout time.Duration) erro
 	return nil
 }
 
+// deleteNamespaces deletes all namespaces that match the given delete and skip filters.
+// Filter is by simple strings.Contains; first skip filter, then delete filter.
+// Returns the list of deleted namespaces or an error.
+func deleteNamespaces(c *client.Client, deleteFilter, skipFilter []string) ([]string, error) {
+	By("Deleting namespaces")
+	nsList, err := c.Namespaces().List(labels.Everything(), fields.Everything())
+	Expect(err).NotTo(HaveOccurred())
+	var deleted []string
+	var wg sync.WaitGroup
+OUTER:
+	for _, item := range nsList.Items {
+		if skipFilter != nil {
+			for _, pattern := range skipFilter {
+				if strings.Contains(item.Name, pattern) {
+					continue OUTER
+				}
+			}
+		}
+		if deleteFilter != nil {
+			for _, pattern := range deleteFilter {
+				if strings.Contains(item.Name, pattern) {
+					break
+				}
+				continue OUTER
```

Comment:
looking at this again, I think this logic is wrong.

suppose deleteFilter is `["a", "b"]`, and the list of namespaces is `["b"]`. this will incorrectly skip deleting `["b"]`, since it only looks at the first element of `deleteFilter`.

probably this `continue` should be outside of the inner loop?